components/gitpod-db/src/typeorm/auth-provider-entry-db-impl.ts

```diff
@@ -88,6 +88,7 @@ export class AuthProviderEntryDBImpl implements AuthProviderEntryDB {
         const query = repo
             .createQueryBuilder("auth_provider")
             .where(`auth_provider.ownerId = :ownerId`, { ownerId })
+            .andWhere("(auth_provider.organization_id IS NULL OR auth_provider.organization_id = '')")
```

Comment:
I think it's the best way to filter at DB query level, like you did here.

`markAsVerified` would be broken for Org-level providers with that change expectedly. A simple fix would bo to add `orgId` and query the list of providers accordingly.